Stun warfare rebalance - CBT edition

DeltaFire15 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Changes up the whole dynamic of a possible fight of the two cults by removing stun mechanics and replacing them with mainly debuffings.

This means (only against hostile cultists), instead of hardstunning, they now some different things:

Clockwork spearthrow now confuses, deals extra burn damage and slightly sets on fire + does a very weak knockdown

Blood cult spearthrow now confuses, deals extra brute damage and does a light knockdown.

Clock cult kindle now confuses, applies stuttering and makes drowsy (only the confusion effect can stack) and deals some burn + applies belligerent (aka toggles movement to walk and deals damage if you try to run)

Blood cult stunhand now confuses, applies stuttering and makes drowsy aswell plus deals some brute (with only confusion able to stack aswell), but the debuff itself lasts a bit longer since they don't get belligerent.

And finally, the clockie transgression / stun sigil now instead of hardstunning harder deals burn, does a light knockdown and sets them on fire.

I am open to suggestions like if something seems to strong / weak.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Currently, the dynamic of a cult vs cult battle is just: Use the cults stun > stabby stab > win, making it very unenjoyable in general because noone likes instawin hardstuns. This PR aims to change that up and to make cult vs. cult a bit more enjoyable and more worthy of being adminbussed in.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
balance: Rebalanced cult vs cult stun effects to debuff instead of stun
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DeltaFire15 committed:
Merge pull request #2 from Citadel-Station-13/master

master hardsync

DeltaFire15 committed:
Cult vs Cult changes, Ratvar side

Changes the hardstuns by kindle and spearthrow into debuffs if attacking hostile (Nar'Sian) cultists

DeltaFire15 committed:
Some more changes to clock stuff

Adds some flavor for the kindle spell, changes the transgression sigil to burn cultists instead of stunning, adds belingerent debuff to spearthrow and kindle attack when used on hostile cultists

DeltaFire15 committed:
Little typofix

Bruh

DeltaFire15 committed:
Changes blood halberd to no longer hardstun hostile cult when thrown

It will now, simillar to the Ratvarian counterpart, instead apply damage and some confusion, plus some weak knockdown.

DeltaFire15 committed:
Changes up the blood stunhand against hostile cultists

The stunhand used by bloodcult will no longer hardstun + deal damage to hostile cultists. Instead, it will deal some brute + debuff the enemy cultist, duration of the debuffing is a bit longer then for the clockie kindle since that one also applies belligerent.
Also fixes up some stuff / adds a safety check in the clockie changes

DeltaFire15 committed:
Sync with master (#3)

Alot of stuff, honk

DeltaFire15 committed:
Another sync (#4)

AAAAAAAAA

DeltaFire15 committed:
Merge branch 'master' into cbt-clockvsbloodtorture

DeltaFire15 committed:
Okay epic

Bruh

DeltaFire15 committed:
Merge branch 'master' into cbt-clockvsbloodtorture

DeltaFire15 committed:
Okay cool

DeltaFire15 committed:
Hmmm

DeltaFire15 committed:
Bruh moment

fixes a missing bracketty boi

DeltaFire15 committed:
Just some fixes

Fun fun fun

silicons commented on code/modules/antagonists/clockcult/clock_helpers/slab_abilities.dm:
```diff
@@ -179,6 +179,7 @@
 	. = ..()
 
 /obj/item/projectile/kindle/on_hit(atom/target, blocked = FALSE)
+	var/narsiandog_punish = 8
```
no reason to do this, unnecessary variables.
also down below do something like C.confused += CLAMP(16 - C.confused, 0, 8) to prevent infinite stacking.

silicons commented on code/modules/antagonists/clockcult/clock_items/clock_weapons/ratvarian_spear.dm:
```diff
@@ -56,8 +57,13 @@
 				L.visible_message("<span class='warning'>[src] bounces off of [L], as if repelled by an unseen force!</span>")
 		else if(!..())
 			if(!L.anti_magic_check())
-				if(issilicon(L) || iscultist(L))
```
removing the knockdown entirely is a bit silly maybe make it just knockdown(1) or something so it knocks down without doing much stamina

silicons commented on code/modules/antagonists/cult/blood_magic.dm:
```diff
@@ -409,6 +409,7 @@
 	name = "Stunning Aura"
 	color = RUNE_COLOR_RED
 	invocation = "Fuu ma'jin."
+	var/ratvarscum_punish = 10
```
more serious/at a glance and utilitarian variable names, same for the other places.

silicons commented on code/modules/antagonists/cult/blood_magic.dm:
```diff
@@ -438,21 +439,27 @@
+					var/mob/living/carbon/C = L
+					C.stuttering = max(ratvarscum_punish, C.stuttering)
+					C.drowsyness = max(ratvarscum_punish, C.drowsyness)
+					C.confused += ratvarscum_punish
```
same for above make sure it's clamped

silicons reviewed (dismissed):
comments

DeltaFire15 committed:
Changes code as requested by kev

Very very light knockdown for the clock spearthrow, CLAMPed values for everything that used a += (including the blood stun used on normal people), removed the vars as they were deemed unneccessary by kev so the values were moved directly to the debuff effects.

DeltaFire15 committed:
Whoops

Fixes a typo

Ghommie reviewed (approved):
Cult vs Cult is very rare and normally only happens through admin-fuckery.
Merging this as a testmerge would be useless, it looks fine.
